fieldpath/element_test.go

```diff
@@ -29,3 +29,20 @@ func TestPathElementSet(t *testing.T) {
 		t.Errorf("unequal sets should not equal")
 	}
 }
+
+func TestPathElementSetIterate(t *testing.T) {
+	set := &PathElementSet{}
+	toAdd := 5
+	for i := 0; i < toAdd; i++ {
+		p := i
+		pe := PathElement{Index: &p}
+		set.Insert(pe)
+	}
+	count := 0
+	set.Iterate(func(_ PathElement) {
```

Comment:
(It doesn't hurt to add this but I think this was effectively tested by the test for `String()`?)